toolkit/tools/pkg/osmodifierlib/modifierutils.go

```diff
@@ -23,3 +29,69 @@ func doModifications(baseConfigPath string, osConfig *imagecustomizerapi.OS) err
 
 	return nil
 }
+
+func doModificationsWithoutConfig() error {
+	var dummyChroot safechroot.ChrootInterface = &safechroot.DummyChroot{}
+
+	// Get verity changes in /boot/grub2/grub.cfg
+	verityParams, err := getVerityChanges(dummyChroot)
+	if err != nil {
+		return fmt.Errorf("error getting verity changes:\n%v", err)
+	}
+
+	bootCustomizer, err := imagecustomizerlib.NewBootCustomizer(dummyChroot)
+	if err != nil {
+		return err
+	}
+
+	// Apply the verity changes to /etc/default/grub
+	err = bootCustomizer.ApplyChangesToGrub("GRUB_CMDLINE_LINUX", verityParams)
```

Comment:
I don't really understand what this change is doing. MIC doesn't write to the `/etc/default/grub` file since doing so would break the hash. So, why is it ok to write to the `/etc/default/grub` file here?

Reply:
Trident will be running osmodifier with an overlay mounted over /etc, which means that editing /etc/default/grub will impact only the copy in the overlay and not change the hash